### PR TITLE
Support numeric facets in cell filtering API requests (SCP-5410)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND & CHANGES
This update adds support for numeric facets to cell filtering API requests.  Now, instead of throwing a `400`, the API will add in the actual numeric value for each cell in a numeric annotation.  This will allow for filtering on the front end via a slider bar and a range check.

#### MANUALT TESTING
1. Boot as normal and open the [Swagger documentation](https://localhost:3000/single_cell/api/v1#/Visualization/study_annotation_facets_path) for the `/annotations/facets` endpoint
2. Find the study accession for the synthetic study `Male Mouse brain`
3. In the Swagger doc, enter the accession and the following:
    * `annotations` : `disease__ontology_label--group--study,Intensity--numeric--cluster` 
    * `cluster` : `cluster.tsv`
4. Click "Execute" and confirm that the request completes, and that the response looks like the following:
```
{ 
  "cells": [ [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 1 ], [ 0, 2 ], [ 0, 2 ], [ 0, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 1, 2 ], [ 0, 2 ], [ 0, 2 ], [ 0, 2 ], [ 0, 2 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 3 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 4 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 5 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 6 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 7 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ], [ 0, 8 ] ],
  "facets": [
    {
      "annotation": "disease__ontology_label--group--study",
      "groups": [
        "oral tuberculosis",
        "intestinal tuberculosis"
      ]
    },
    {
      "annotation": "Intensity--numeric--cluster",
      "groups": []
    }
  ]
}
```

### CODE REVIEW

File: app/controllers/api/v1/visualization/annotations_controller.rb [hunk 1]
```diff
@@ -290,16 +286,19 @@ def facets
...
                 label = annotation_arrays[annotation][index] || '--Unspecified--'
               end
-              if label == ''
-                label = '--Unspecified--'
+              if annotation_type == 'group'
+                label = '--Unspecified--' if label.blank?
+
+                facet[:groups].index(label)
+              else
+                label.presence || Float::NaN
```
Comment (eweitz) [inline app/controllers/api/v1/visualization/annotations_controller.rb:300]: Nice use of [`.presence`](https://stackoverflow.com/a/19637602) here!

**Review by eweitz — APPROVED** [review]
Code looks good -- being able to filter cells by QC metrics and other numerical values will be quite useful.

**Comment by bistline** [comment]
Looking at log output shows that upstream instability led to CI failure:
```
FireCloud API request (POST) https://api.firecloud.org/api/workspaces with tracking identifier: 6580c131421aa901ca0b41a6
Timed out reading data from server: ;  encountered when requesting 'https://api.firecloud.org/api/workspaces', attempt #1
FireCloud API request (POST) https://api.firecloud.org/api/workspaces with tracking identifier: 6580c131421aa901ca0b41a6
500 Internal Server Error: {
  "causes": [],
  "exceptionClass": "java.sql.SQLIntegrityConstraintViolationException",
  "message": "Duplicate entry 'single-cell-portal-test-validation-ingest-pipeline-parse-failure' for key 'IDX_WS_UNIQUE_NAMESPACE_NAME'",
  "source": "rawls",
 
```

**Review by jlchang — APPROVED** [review]
Functional review - works as described.

**Pull request merged**